packages/fuels-programs/src/contract.rs

```diff
@@ -175,17 +167,46 @@ impl DeployConfiguration {
 /// compiling, deploying, and running transactions against a contract.
 /// The contract has a wallet attribute, used to pay for transactions and sign them.
 /// It allows doing calls without passing a wallet/signer each time.
-pub struct Contract;
+pub struct Contract {
+    binary: Vec<u8>,
+    salt: Salt,
+    storage_slots: Vec<StorageSlot>,
+    contract_id: ContractId,
+    state_root: Bytes32,
+}
 
 impl Contract {
-    pub fn compute_contract_id_and_state_root(
-        compiled_contract: &CompiledContract,
+    pub fn new(binary: Vec<u8>, salt: Salt, storage_slots: Vec<StorageSlot>) -> Self {
+        let (contract_id, state_root) =
+            Self::compute_contract_id_and_state_root(&binary, &salt, &storage_slots);
+
+        Self {
+            binary,
+            salt,
+            storage_slots,
+            contract_id,
+            state_root,
+        }
+    }
+
+    pub fn contract_id(&self) -> ContractId {
+        self.contract_id
+    }
+
+    pub fn state_root(&self) -> Bytes32 {
+        self.state_root
+    }
+
+    fn compute_contract_id_and_state_root(
```

Comment:
And this would be a method of `CompiledContract`. I do think we can find a better name.